src/control/cmd/agent/mgmt_rpc.go

```diff
@@ -74,7 +76,13 @@ func (mod *mgmtModule) handleGetAttachInfo(reqb []byte) ([]byte, error) {
 
 	log.Debugf("GetAttachInfo %s %v", mod.ap, *req)
 
-	conn, err := grpc.Dial(mod.ap, grpc.WithInsecure())
+	dialOpt, err := security.DialOptionForTransportConfig(mod.tcfg)
+	if err != nil {
+		return nil, err
+	}
+	opts := []grpc.DialOption{dialOpt}
```

Comment:
This isn't necessary, because grpc.Dial() accepts the options as varargs so you could just pass in dialOpt directly. I don't feel strongly about it, but it's a little confusing.

Reply:
AI tried passing the opts directly and it did not work the compiler pushed back on it. I tried both passing both dialOpt and dialOpt... and it only accepted dialOpt wrapped as an array and passed with ...

Reply:
Hmm. I just pulled the PR and built it with dialOpt passed in directly. Not sure what you were seeing but the function signature of Dial is:
```
func Dial(target string, opts ...DialOption) (*ClientConn, error)
```
I'm certain that will take a single DialOption; that's how variadic functions work. In any case -- It's not a blocker for me, just pointing it out for posterity.

Reply:
Huh. I'm not sure what the issue was as well then. I definitely had some issue with that but we can change it. Might also be nice to leave that way so we can add more values to opts if we need to in the future.